WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,102 @@
 <% provide(:title, 'Competitions') %>
 
 <div class="container">
-  <h1><%= yield(:title) %></h1>
-
-  <%= link_to "New competition", new_competition_path %>
-  <table class="table">
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <div id="scroller-anchor"></div>
+  <div id="scroller">
+    <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+      <%= render 'index_form' %>
+    <% end %>
+  </div>
+    <% if @competitions.length == 0 %>
+      <div class="alert alert-warning">No competitions found!</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <div class="row competition-list">
+          <div class="col-md-12">
+            <%= render 'index_table', competitions: @in_progress_competitions, title: "Competitions in progress" %>
+          </div>
+
+          <div class="col-md-6">
+            <%= render 'index_table', competitions: @upcoming_competitions, title: "Upcoming competitions" %>
+          </div>
+          <div class="col-md-6">
+            <%= render 'index_table', competitions: @past_competitions, title: "Past competitions" %>
+          </div>
+        </div>
+      <% else %>
+        <div class="row competition-list">
+          <div class="col-xs-12 col-md-12">
+            <div id="map"></div>
+          </div>
+        </div>
+      <% end %>
     <% end %>
-  </table>
 </div>
+
+<script>
+function moveScroller() {
+    var move = function() {
+        var st = $(window).scrollTop();
+        var ot = $("#scroller-anchor").offset().top;
+        var s = $("#scroller");
+        if(st > ot) {
+            s.css({
+                position: "fixed",
+                top: "0px",
+                "z-index": 10
+            });
+        } else {
+            if(st <= ot) {
+                s.css({
+                    position: "relative",
+                    top: ""
+                });
+            }
+        }
+    };
+    $(window).scroll(move);
+    move();
+}
+
+function mapSize() {
+  var nav = parseInt($(".navbar").outerHeight(true));
+  var head = parseInt($("h2").outerHeight(true));
+  var form = parseInt($(".form-group").outerHeight(true));
+  var foot = parseInt($(".footer").outerHeight(true));
+
+  var height = parseInt($(window).innerHeight());
+
+  var map = height - nav - head - form - foot;
+
+  $("#map").height( map );
+}
+
+$(window).resize(function() {
+  mapSize();
```

Comment:
Moved it inside of the <% if params[:display] ...
and added the condition to window resize.